### ISSUE TITLE
Shuffle flag is true in make_dataset function

### ISSUE THREAD

**Issue by hemantnyadav** [tensorflow/tensorflow#60589]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Support

### Have you reproduced the bug with TF nightly?

No

### Source

source

### Tensorflow Version

2.0

### Custom Code

Yes

### Current Behaviour?

Just replicating an example of Time Series forecasting from link [TSF](https://www.tensorflow.org/tutorials/structured_data/time_series#recurrent_neural_network)

There is a function make_dataset which uses timeseries_dataset_from_array.
The function timeseries_dataset_from_array shuffle flag is set to true which is not allowed in time series forecasting.

### Standalone code to reproduce the issue

```shell
def make_dataset(self, data):
  data = np.array(data, dtype=np.float32)
  ds = tf.keras.utils.timeseries_dataset_from_array(
      data=data,
      targets=None,
      sequence_length=self.total_window_size,
      sequence_stride=1,
      shuffle=True,
      batch_size=32,)

  ds = ds.map(self.split_window)

  return ds

WindowGenerator.make_dataset = make_dataset
```

### Relevant log output

_No response_</details>

**Comment by SuryanarayanaY** [comment]
Hi @hemantnyadav ,

thanks for reaching us. I believe shuffle=True can be used for some cases. But before that we might need to split the dataset first to train,test/validation datasets. Here `shuffle=True` means it will shuffle means only the rows will be shuffled but not the columns(Features).

Suppose if we have some time series data and we want to make a dataset such that it will choose 6 data points in sequence and 5 among them are as inputs and 6th one as label. Here we are trying to predict some output based on previous 5 time stamps. Please note that here the 5 time stamps will be in order only i.e columns(Features) are not being shuffled. But we are not establishing relationship between one data point to other but rather we are trying to create relationship between the same datapoint(sequence of 5 timesteps with 6th one as output).Sometimes we may need to shuffle the data in terms of batches only. We should carefully choose `shuffle=True` based on use case.

Please refer to some more web resources for better understanding [link1](https://stackoverflow.com/questions/70326567/shuffling-a-classification-timeseries-data), [link2](https://datascience.stackexchange.com/questions/54237/is-it-valid-to-shuffle-time-series-data-for-a-prediction-task) and [link3](https://www.analyticsvidhya.com/blog/2021/06/deep-dive-into-time-series-data-with-single-neuron/).

Thanks!

**Comment by hemantnyadav** [comment]
Thank you for your reasonable explanation.
but my question is specifically indicating the response you have given is not mentioned in the article.
As the article is for an audience who wants to get started with it or someone who wants to create a testbed for such an experiment.
In such cases, they need to verify it with original data rather than relying on API-based error rate.
When the sequence of data points (not features but rows) changes, verification of results manually becomes very difficult. So it would be great if you incorporated these details about shuffle flag in the article itself.
Thank you.

**Comment by SuryanarayanaY** [comment]
Hi @hemantnyadav,

Documentation is not exhaustive and I agree that. Because the concept is general only and not specific to Tensorflow and hence it might not be documented specifically.Sometimes adding exhaustive documentation for general concepts may gives feeling of overloaded documentation.Hence it might have ignored.If you still feel adding it might be useful then will see what can be done here.

Thanks!
